protocol/tests/test_protocol.py

```diff
@@ -0,0 +1,149 @@
+import pytest
+import math
+import ctypes
+
+
+@pytest.mark.parametrize(
+    "test_data",
+    [
+        {"a": ["foo", "bar"], "b": ["baz", "qux"]},
+        {"a": [1.5, 2.5, 3.5], "b": [9.2, 10.5, 11.8]},
+        {"A": [1, 2, 3, 4], "B": [1, 2, 3, 4]},
+    ],
+    ids=["str_data", "float_data", "int_data"],
+)
+def test_only_one_dtype(test_data, df_from_dict):
+    columns = list(test_data.keys())
+    df = df_from_dict(test_data)
+    dfX = df.__dataframe__()
+
+    column_size = len(test_data[columns[0]])
+    for column in columns:
+        assert dfX.get_column_by_name(column).null_count == 0
+        assert dfX.get_column_by_name(column).size == column_size
+        assert dfX.get_column_by_name(column).offset == 0
+
+
+def test_float_int(df_from_dict):
+    df = df_from_dict(
+        {
+            "a": [1, 2, 3],
+            "b": [3, 4, 5],
+            "c": [1.5, 2.5, 3.5],
+            "d": [9, 10, 11],
+            "e": [True, False, True],
+            "f": ["a", "", "c"],
+        }
+    )
+    dfX = df.__dataframe__()
+    columns = {"a": 0, "b": 0, "c": 2, "d": 0, "e": 20, "f": 21}
+
+    for column, kind in columns.items():
+        colX = dfX.get_column_by_name(column)
+        assert colX.null_count == 0
+        assert colX.size == 3
+        assert colX.offset == 0
+
+        assert colX.dtype[0] == kind
+
+    assert dfX.get_column_by_name("c").dtype[1] == 64
+
+
+def test_na_float(df_from_dict):
+    df = df_from_dict({"a": [1.0, math.nan, 2.0]})
+    dfX = df.__dataframe__()
+    colX = dfX.get_column_by_name("a")
+    assert colX.null_count == 1
+
+
+def test_noncategorical(df_from_dict):
+    df = df_from_dict({"a": [1, 2, 3]})
+    dfX = df.__dataframe__()
+    colX = dfX.get_column_by_name("a")
+    with pytest.raises(TypeError):
+        colX.describe_categorical
+
+
+def test_categorical(df_from_dict):
+    df = df_from_dict(
+        {"weekday": ["Mon", "Tue", "Mon", "Wed", "Mon", "Thu", "Fri", "Sat", "Sun"]},
+        is_categorical=True,
+    )
+
+    colX = df.__dataframe__().get_column_by_name("weekday")
+    is_ordered, is_dictionary, _ = colX.describe_categorical
+    assert isinstance(is_ordered, bool)
+    assert isinstance(is_dictionary, bool)
+
+
+def test_dataframe(df_from_dict):
+    df = df_from_dict(
+        {"x": [True, True, False], "y": [1, 2, 0], "z": [9.2, 10.5, 11.8]}
+    )
+    dfX = df.__dataframe__()
+
+    assert dfX.num_columns() == 3
+    assert dfX.num_rows() == 3
+    assert dfX.num_chunks() == 1
+    assert dfX.column_names() == ["x", "y", "z"]
+    assert (
+        dfX.select_columns((0, 2)).column_names()
+        == dfX.select_columns_by_name(("x", "z")).column_names()
+    )
```

Comment:
[`.column_names()` returns an iterator](https://data-apis.org/dataframe-protocol/latest/API.html#:~:text=Return%20an%20iterator%20yielding%20the%20column%20names.), so it should probably be converted to a list before comparison
```suggestion
    assert list(dfX.column_names()) == ["x", "y", "z"]
    assert (
        list(dfX.select_columns((0, 2)).column_names())
        == list(dfX.select_columns_by_name(("x", "z")).column_names())
    )
```

Reply:
@dchigarev I changed this